### PR TITLE
Backlog 15618 Generate index in default language only

### PR DESCRIPTION
**Pull request by AKarmanov**
## JIRA

https://jira.jahia.org/browse/BACKLOG-15618

## Description

Generate sitemap.index in default language only, fix paths when with context

### CODE REVIEW

File: src/main/resources/jseont_sitemap/xml/sitemap.index.jsp [hunk 1]
```diff
@@ -42,24 +43,21 @@
...
+                                        <loc>${url.server}${languageResource}</loc>
                                 </sitemap>
                         </c:if>
                 </c:forEach>
 
                 <%--  Separate sitemaps for jseont:sitemapResource node option --%>
-                <%-- TODO: replace jnt:page to jseont:sitemapResource once nodetype is available --%>
                 <jcr:jqom var="additionalMaps">
-                        <query:selector nodeTypeName="jnt:page" selectorName="stmp"/>
+                        <query:selector nodeTypeName="jseont:sitemapResource" selectorName="stmp"/>
```
Comment (gflores-jahia) [inline src/main/resources/jseont_sitemap/xml/sitemap.index.jsp:55]: missing a check to add nodes only when parent `jseomix:sitemapResource` has `createSitemap` attribute enabled

File: src/main/resources/jseont_sitemap/xml/sitemap.index.jsp [hunk 2]
```diff
@@ -18,10 +18,11 @@
 <%--@elvariable id="workspace" type="java.lang.String"--%>
 
 <c:set target="${renderContext}" property="contentType" value="text/xml;charset=UTF-8"/>
+
 <?xml version="1.0" encoding="UTF-8"?>
 <sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
-        <c:if test="${renderContext.liveMode}">
-                <c:set var="siteMapPath" value="${fn:replace(currentNode.url, '.html', '')}" />
+        <c:if test="${renderContext.liveMode and renderContext.site.defaultLanguage eq renderContext.mainResourceLocale.language}">
```
Comment (gflores-jahia) [inline src/main/resources/jseont_sitemap/xml/sitemap.index.jsp:24]: does jsp's support `and` or does that have to be `&&`?

**Review by gflores-jahia — CHANGES_REQUESTED** [review]

File: src/main/resources/jseont_sitemap/xml/sitemap.index.jsp (hunk 2 above)
Comment (AKarmanov) [reply src/main/resources/jseont_sitemap/xml/sitemap.index.jsp:24]: it supports both

File: src/main/resources/jseont_sitemap/xml/sitemap.index.jsp (hunk 1 above)
Comment (AKarmanov) [reply src/main/resources/jseont_sitemap/xml/sitemap.index.jsp:55]: Yes, you're right!

File: src/main/resources/jseont_sitemap/xml/sitemap.index.jsp (hunk 2 above)
Comment (gflores-jahia) [reply src/main/resources/jseont_sitemap/xml/sitemap.index.jsp:24]: learned something new today :)

**Review by gflores-jahia — APPROVED** [review]

**Pull request merged**